text/0000-export-to-oci.md

```diff
@@ -0,0 +1,261 @@
+
+```=shell
+cd oci
+> tree .
+.
+└── oci/
+    └── my-app-image/
+        ├── blobs/
+        │   └── sha256/
+        │       ├── 65d9067f915e01...tar
+        │       ├── 6905011516dcf4...tar
+        │       └── 83d85471d9f8a3...tar
+        ├── index.json
+        └── oci-layout
+```
+
+### Exporting to a Registry
+
+```=shell
+> export CNB_LAYOUT=oci
+> /cnb/lifecycle/exporter gcr.io/my-repo/my-app-image
+```
+
+Or
+
+```=shell
+>  /cnb/lifecycle/exporter -layout oci gcr.io/my-repo/my-app-image
+```
+
+The expected output is the `my-app-image` exported in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format into the `/oci/` folder
+
+```=shell
+cd oci
+> tree .
+.
+└── oci/
+    └── my-app-image/
+        ├── blobs/
+        │   └── sha256/
+        │       ├── 65d9067f915e01...tar
+        │       ├── 6905011516dcf4...tar
+        │       └── 83d85471d9f8a3...tar
+        ├── index.json
+        └── oci-layout
+
+```
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The lifecycle phases affected by this new behavior is: [Export](https://buildpacks.io/docs/concepts/components/lifecycle/export/)
+
+At high level view the propose solution can be summarize with the following container diagram from the C4 model
+
+![](https://i.imgur.com/0OLSK8o.png)
+
+
+Notice that we are relying on the OCI format Specification to expose the data for `Platforms`
+
+The following new input is proposed to be added to this phase
+
+| Input             | Environment Variable  | Default Value            | Description
+|-------------------|-----------------------|--------------------------|----------------------
+| `<layout>`      |  `CNB_LAYOUT_DIR` | "" | The root directory where the OCI image will be written. The presence of a none empty value for this environment variable will enable the feature. |
+
+Let's see the propose flow
+
+```mermaid
+flowchart
+  A{"IS -layout OR
+  CNB_LAYOUT_DIR
+  defined?"} -->|Yes| B
+  A -->|No| END
+  B{"IS -launch-cache
+  defined?"} -->|Yes|D
+  B -->|No| E
+  E{"DOES
+  layout-dir/image
+  exist?"} --> |Yes| L
+  L[...]
+  E --> |No| M
+  M[Create layout-dir/image directory] --> O[export-dir = layout-dir/image]
+  O --> I
+  D[/Warn: will export to launch cache dir/] --> F
+  F{"DOES
+  launch-cache/image
+  dir exist?"} -->|Yes| G
+  G[ ...]
+  F -->|No| H
+  H[Create launch-cache/image directory] --> N[export-dir = launch-cache/image]
+  N --> I[Write image to $export-dir in OCI format **]
+  I --> J[Calculate manifest's digest]
+  J --> K[/Write digest into report.toml/]
+  K -->END((End))
+```
+
+Notes:
+  - WHEN `-launch-cache` flow is executed
+    - The content of `blobs/<alg>/<encoded>` MAY contain symbolic links to content saved in the launch cache to avoid duplicating files.  
+    - The content of `blobs/<alg>/<encoded>` MAY reference tar files in **uncompressed** format because that's how they are saved in the cache
```

Comment:
We should probably change this to be consistent with the layout/registry case. We could uncompress the blobs when we need to save them into the daemon. We should probably understand the performance impact of this and whether simplicity is worth the tradeoff.

Reply:
My concern here is:
- Currently, the tar files are saved **uncompressed** in the launch cache. Was it done by design for some reason? I don't want to propose a breaking change without having the whole context of why it was done like this in the first place.
- I agree with you, the main reason is performance, maybe playing a little bit with the code I will get some hint about performance impact.